schema/schema.go

```diff
@@ -13,6 +13,18 @@ import (
 	"gorm.io/gorm/logger"
 )
 
+const (
```

Comment:
You may introduce a new type like below:

```suggestion
type CallbackMethodType string

const (
    callbackBeforeCreate CallbackMethodType = "BeforeCreate"
    ...
```

Reply:
Sure, I can do this. This was in fact the original plan.

 It does mean lots of string() casting everywhere.